### ISSUE TITLE
Redshift database and TABLESAMPLE BERNOULLI 

### ISSUE THREAD

**Issue by rusq** [tokern/piicatcher#118]
Redshfit database does not support TABLESAMPLE BERNOULLI, which piicatcher issues.

Command used to invoke:
```
piicatcher db -s xxxxxx.redshift.amazonaws.com -R 5439 -u xxxxx -p xxxxx -d etl --connection-type redshift -c deep -n ads -t dim_account
```

Resulting error:
```

Traceback (most recent call last):
  File "/Users/<username>/.virtualenvs/piicatcher/bin/piicatcher", line 8, in <module>
    sys.exit(cli())
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/piicatcher/explorer/databases.py", line 140, in cli
    RelDbExplorer.dispatch(ns)
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/piicatcher/explorer/explorer.py", line 59, in dispatch
    explorer.scan()
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/piicatcher/explorer/explorer.py", line 87, in scan
    schema.scan(self._generate_rows)
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/piicatcher/explorer/metadata.py", line 73, in scan
    child.scan(generator)
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/piicatcher/explorer/metadata.py", line 116, in scan
    for row in generator(
  File "/Users/<username>/.virtualenvs/piicatcher/lib/python3.8/site-packages/piicatcher/explorer/explorer.py", line 171, in _generate_rows
    cursor.execute(query)
psycopg2.errors.SyntaxError: syntax error at or near "BERNOULLI"
LINE 1: ...r_id" from ads.dim_account_attribute TABLESAMPLE BERNOULLI ...
```

**Comment by vrajat** [comment]
I have a potential fix in a branch. Will you be open to trying it? I can give you instructions to install and run.

I dont have access to a redshift instance to test my changes at the moment.

**Comment by rusq** [comment]
Sure

**Comment by vrajat** [comment]
Thanks! Instructions:

    git clone https://github.com/vrajat/piicatcher -b issue-118
    cd piicatcher
    pipenv install -e .
    pipenv shell
    piicatcher ....

**Comment by rusq** [comment]
I don't seem to see the branch, @vrajat  did you push it?

```
[0:~/wf/piicatcher git:(master)> git branch -a | cat
* master
  remotes/origin/HEAD -> origin/master
  remotes/origin/column_name_pii
  remotes/origin/files
  remotes/origin/iss-36
  remotes/origin/iss-76
  remotes/origin/iss-90
  remotes/origin/issue_22
  remotes/origin/master
  remotes/origin/orm
  remotes/origin/snowflake
```

**Comment by vrajat** [comment]
Fixed. Can you check again?

**Comment by rusq** [comment]
Still running into the same issue, but this is due to `factory` not returning the correct explorer
see this line:
https://github.com/vrajat/piicatcher/blob/issue-118/piicatcher/explorer/databases.py#L139

After removing `or 'redshift` part it worked!

**Comment by rusq** [comment]
P.S I wasn't able to make this version going with pipenv, I always have problems with pipenv - it doesn't like me and I don't like it, so I used the "production" setup with importing the (patched version of the) module from the current directory.

**Comment by rusq** [comment]
Made a suggestion on the PR #120

**Comment by vrajat** [comment]
*sigh* - Silly mistake. Thanks for the suggestion and trying out the fix. I have released v0.12.3 with the fix. Also please take this [survey](https://forms.gle/Ns6QSNvfj3Pr2s9s6) It helps me understand what developers need from this project.

**Comment by rusq** [comment]
No problem, thanks for fixing it so fast, glad I was able to help.

We are experimenting with piicatcher at the moment, I'll be able to answer the survey at the end of January, right now I don't have anything concrete to say, but I'll make a note of that to myself - will be happy to provide feedback in future
